src/console/dlt-control-common.c

```diff
@@ -565,12 +565,18 @@ int dlt_control_send_message(DltControlMsgBody *body, int timeout)
         pr_error("Sending message to daemon failed\n");
         dlt_message_free(msg, get_verbosity());
         free(msg);
+
+        /* make sure the mutex is unlocked to prevent deadlocks */
+        pthread_mutex_unlock(&answer_lock);
         return -1;
     }
 
-    /* If we timeout the lock is not taken back */
-    if (!pthread_cond_timedwait(&answer_cond, &answer_lock, &t))
-        pthread_mutex_unlock(&answer_lock);
+    /*
+    * When a timeouts occurs, pthread_cond_timedwait()
```

Comment:
this comment is referencing the posix specification as the manpage on pthread_cond_timedwait is not 100% clear on this